src/terminal.py

```diff
@@ -241,13 +242,41 @@ async def _update(self):
             self.new_events
             and self.terminal_prefix is not None
             and has_events_with_string(
-                self.new_events, self.terminal_prefix, self.prompt_buffer
+                self.new_events, self.terminal_prefix, self.prompt_buffer + 1
             )
         ):
             return
 
+        # Find the index of the (N+1)th prompt (we want to send everything up to but not
+        # including this prompt)
+        prompt_indices = [
+            i
+            for i, event in enumerate(self.new_events)
+            if self.terminal_prefix in event[2]
```

Comment:
I'm a bit concerned about someone having the terminal prefix in output, for whatever reason, but at this point it's probably fine to not handle that